src/Error/Middleware/ErrorHandlerMiddleware.php

```diff
@@ -72,18 +75,25 @@ class ErrorHandlerMiddleware implements MiddlewareInterface
     /**
      * Constructor
      *
-     * @param \Cake\Error\ErrorHandler|null $errorHandler The error handler.
-     * @param array $config Configuration options for error handler.
-     *   These options will be ignored if an ErrorHandler instance is passed as
-     *   first argument.
+     * @param \Cake\Error\ErrorHandler|array $errorHandler The error handler instance
+     *  or config array.
      */
-    public function __construct($errorHandler = null, array $config = [])
+    public function __construct($errorHandler = [])
     {
-        $this->setConfig($config);
+        if (is_array($errorHandler)) {
```

Comment:
If you wanted to allow a more graceful upgrade process we could use `func_get_args()` to allow the second parameter to continue being used and emit a deprecation warning? Perhaps its not a big deal though as we've not finalized everything yet.

Reply:
Done